### ISSUE TITLE
[question] conan 2.0 migration path for `env_info` and `user_info`

### ISSUE THREAD

**Issue by prince-chrismc** [conan-io/conan#12334]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

This came up in https://github.com/conan-io/conan-center-index/issues/13503, but it's also in a few other recipes

Currently in Conan 1.x we needed to use the `env_info` and `user_info` these are removed in Conan 2.0 and result is hard errors when they are present. https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-package-info-method.

In ConanCenter, this means we need to have both the old and new while down stream recipes are not yet updated. so we are adding a check for the `conan_version` to ensure the problematic code is not called in the new version.

https://github.com/conan-io/conan-center-index/blob/412a59575357fb352b14caa29312263c4081897e/recipes/strawberryperl/all/conanfile.py

```py
from conan import ConanFile, conan_version

    def package_info(self):
        # TODO remove once conan v2 is the only support and recipes have been migrated
        if Version(conan_version) < "2.0.0-beta":
            bin_path = os.path.join(self.package_folder, "bin")
            self.env_info.PATH.append(bin_path)
            self.user_info.perl = perl_path
```

**Comment by czoido** [comment]
Thanks @prince-chrismc for reporting this. I think we could:

- Implement `env_info` and `user_info` as fake attributes in Conan 2.0 so the error is not thrown
- [Output a warning message](https://github.com/conan-io/conan/issues/12345) in 2.0 to inform users that those attributes are deprecated (we should also include other things like .names that are still there to provide compatibility). This is in order to reduce the risk of breaking users when we pass from Conan 2.0 to 2.X

I'm marking this for the next 2.0-beta5

**Comment by uilianries** [comment]
Okay, so we no longer need to check Conan version + env_info, correct?

Asking because PRs in CCI started to use it and I would like to avoid asap.

**Comment by czoido** [comment]
@prince-chrismc I'm closing this as https://github.com/conan-io/conan/pull/12351 was merged (will be released in beta5) and now Conan 2.0 recipes won't fail if they have `env_info` and `user_info` although it will have no effect at all

**Comment by prince-chrismc** [comment]
🚀
